### PR TITLE
[#4617] Add information about versioning external test data.

### PR DESCRIPTION
**Pull request by bgola**
Scope
=====

GECK should talk about versioning external data/ test fixtures (including usernames / passwords), ip addresss.

Changes
=======

Added some example of how to version external automated test data.

As a drive by I have fixed a few warnings/syntax issues on the geck.

How to try and test the changes
===============================

reviewers: @adiroiban @hcs0

### CODE REVIEW

**Comment by bgola** [comment]
needs-review

File: docs/programming/testing.rst [hunk 1]
```diff
@@ -99,17 +99,20 @@ same versioning system as the test code.
...
+
+For example when you want to run an end to end test to check that you can
 read and write a document from a database, you will have to start with testing
 the read operation against an existing data.
 
-In this case make sure that all pre-generated data is versioned using a well
-defined convention and that the pre-generated data is never modified or
-removed.
-In this way you can still run older versions of the code.
+In this case make sure that all pre-generated data is versioned never modified
```
Comment (hcs0) [inline docs/programming/testing.rst]: `versioned, never`

Comment (adiroiban) [reply docs/programming/testing.rst]: maybe 


```
All testing data which is controlled outside of the tests for which is not directly generated by the test should be immutable.
Once a branch using the external data is merged into master, the external data should never be changed.
"""

File: docs/programming/testing.rst [hunk 2]
```diff
@@ -99,17 +99,20 @@ same versioning system as the test code.
...
+or removed.
+
 Pre-generated data include pre-generated access to the outside resource
-(username / password / server address / port)
+(username / password / server address / port). For usernames make sure to use
```
Comment (adiroiban) [inline docs/programming/testing.rst]: I think that we should not do only "username", but any persistent configuration.

Username, hostnames, folder-names, table-names... etc.

The fact that using a version as a suffic can be on a separate paragraph.

explain that we version this data and each time we need to update a test which runs with changed data, we create a new external resource with -vN+1 and leave the existing one in place

**Pull request merged**